.github/workflows/build.yml

```diff
@@ -11,66 +11,108 @@ on:
   schedule:
   - cron: '0 0 * * 0' # weekly
 
+env:
+  CI_OS_NAME: linux
+  CI_COMMIT: ${{ github.sha }}
+  CCACHE_COMPRESS: 1
+  CCACHE_DIR: ${{ github.workspace }}/.ccache
+  CCACHE_MAXSIZE: 2Gi  # 2GiB for clang and gcc, 4GiB in total
+
 jobs:
 
-  vlt:
+  Build:
     strategy:
       fail-fast: false
       matrix:
-        include:
+        debug: [ no, yes ]
+        compiler:
           - { cc: clang, cxx: clang++ }
           - { cc: gcc,   cxx: g++     }
     runs-on: ubuntu-20.04
-    name: Ubuntu 20.04 | ${{ matrix.cc }} | build-test
+    name: Build | ${{ matrix.compiler.cc }} | ${{ matrix.debug }}
     env:
-      CI_OS_NAME: linux
-      CI_COMMIT: ${{ github.sha }}
-      CCACHE_COMPRESS: 1
-      CCACHE_DIR: ${{ github.workspace }}/.ccache
-      CCACHE_MAXSIZE: 2Gi  # 2GiB for clang and gcc, 4GiB in total
-      CC: ${{ matrix.cc }}
-      CXX: ${{ matrix.cxx }}
+      CC: ${{ matrix.compiler.cc }}
+      CXX: ${{ matrix.compiler.cxx }}
+      CI_BUILD_STAGE_NAME: build
     steps:
 
-      - name: Checkout
-        uses: actions/checkout@v2
-
-      - name: Cache
-        uses: actions/cache@v2
-        env:
-          cache-name: ccache
-        with:
-          path: ${{ github.workspace }}/.ccache
-          key: ubuntu-20.04-${{ matrix.cc }}-${{ env.cache-name }}-${{ github.sha }}
-          restore-keys: |
-            ubuntu-20.04-${{ matrix.cc }}-${{ env.cache-name }}
-
-      - name: Install packages for build
-        env:
-          CI_BUILD_STAGE_NAME: build
-        run: bash ci/ci-install.bash
-
-      - name: CCACHE maintenance
-        run: mkdir -p $CCACHE_DIR && bash ci/ci-ccache-maint.bash
-
-      - name: Build
-        env:
-          CI_BUILD_STAGE_NAME: build
-        run: bash ci/ci-script.bash
-
-      - name: Install packages for tests
-        env:
-          CI_BUILD_STAGE_NAME: test
-        run: bash ci/ci-install.bash
-
-      - name: Test
-        env:
-          CI_BUILD_STAGE_NAME: test
-        run: |
-          for item in dist-vlt-0 dist-vlt-1 vltmt-0 vltmt-1; do
-            echo "::group::${item}"
-
-            TESTS="${item}" bash ci/ci-script.bash
-
-            echo '::endgroup::'
-          done
+    - name: Checkout
+      uses: actions/checkout@v2
+
+    - name: Cache
+      uses: actions/cache@v2
+      env:
+        cache-name: ccache
+      with:
+        path: ${{ github.workspace }}/.ccache
+        key: ubuntu-20.04-${{ matrix.compiler.cc }}-${{ env.cache-name }}-${{ github.sha }}
```

Comment:
Probably should use some variable in place of ubuntu-20.04 e.g. the runs-on setting so code is more portable.  Note also we used to run under other versions, we should run under whatever oldest ubuntu is supported too.

I would be tempted to remove github.sha since then we could share cross-commits as most will hit.

Reply:
Done. Now Ubuntu 18.04 and Ubuntu 20.04 are used and the SHA is removed from the cache key.